22_configuration_management/38_technique_editor.txt

```diff
@@ -0,0 +1,85 @@
+=== Technique editor
+
+==== Introduction
+
+===== First, what is a Technique ?
+A technique is a description in code form of what the agent has to do on the node. This code is actually composed of a series of Generic method calls. These different Generic method calls are conditional.
+
+===== What is a Generic method?
+A generic method is a description of an elementary state independent of the operating system (ex: a package is installed, a file contains such line, etc...).
+Generic methods are independent of the operating system (It has to work on any operating system).
+Generic methods calls are conditioned by class expressions, which are boolean expression combining basic conditions with classic boolean operators (ex : operating system is Debian, such generic method produced a modification, did not produce any modification, produced an error, etc…)
+
+
+==== Technique Editor
+
+===== Utility
+Rudder provides a set of pre-defined Techniques that cover some basic configuration and system administration needs. Of course,this set of techniques cannot responds to all of the specific needs of each client. That’s why Rudder integrate the *Technique editor*, a tool to create advanced Techniques.
+Directly accessible from Ruder menu (_Utilities > Technique editor_), this tool has an easy-to-use interface, which doesn’t require any programming skills but nevertheless allows to create complex Techniques.
+
+===== Interface
+Here is an overview of its interface :
+
+image::./images/technique_editor/1-rudder-technique-editor.png[]
+The interface is divided into 3 columns:
+
+
+.1. A column listing custom Techniques
+image::./images/technique_editor/2-list-techniques.png[]
+Here, we can see our previously created Techniques. We can click on them to see their details/edit them, or create a new one by clicking on the “New” button. Theses Techniques are visible in the *ncf techniques* category in the *Active Techniques tree*, so can be used to create new Directives.
```

Comment:
Maybe we should not talk about the "active techniques tree" but only in the "Directive tree" ?